sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -115,6 +116,26 @@ case class CreateTableLikeCommand(
       CatalogTableType.EXTERNAL
     }
 
+    // We only copy source tbl properties if the format is the same with each other
+    val needCopyProperties =
+      (provider.isEmpty || provider == sourceTableDesc.provider) &&
```

Comment:
Why not just compare `newProvider` with `sourceTableDesc.provider`?

When `provider.isEmpty` is true, it covers view case, Hive provider, do we need to copy properties for all these cases?

Reply:
`newProvider` checked `VIEW` and always used `parquet` as the provider.

We only copy properties for tables,  and already checked at below code
```
val newProperties = sourceTableDesc.tableType match {
      case MANAGED | EXTERNAL if needCopyProperties =>
        // Hive only retain the useful properties through serde class annotation.
        // For better compatible with Hive, we remove the metastore properties.
        sourceTableDesc.properties -- DDLUtils.METASTORE_GENERATED_PROPERTIES ++ properties
      case MANAGED | EXTERNAL =>
        properties
      case VIEW =>
        // For view, we just use new properties
        properties
      case other =>
        throw new IllegalArgumentException(
          s"Unknown table type is found at createTableLikeCommand: $other")
    }
```

Reply:
Another case is `provider.isDefined` is false and `fileFormat.inputFormat.isDefined` is true. We might copy properties for it if `fileFormat.inputFormat == sourceTableDesc.storage.inputFormat`. Is it valid to copy properties for this case? When the source provider is different to Hive provider?

Reply:
`inputFormat` only used in Hive provider right ? it will not happen.

Reply:
We already have tests for all the case described above?

Reply:
ok, sounds good.

Reply:
+1 for @maropu's comment. If we have tests for them, it should be good.

Reply:
added more test.